homeassistant/components/rfxtrx/__init__.py

```diff
@@ -179,8 +174,16 @@ async def async_setup_internal(hass: HomeAssistant, entry: ConfigEntry) -> None:
     @callback
     def async_handle_receive(event: rfxtrxmod.RFXtrxEvent) -> None:
         """Handle received messages from RFXtrx gateway."""
-        # Log RFXCOM event
-        if not event.device.id_string:
+
+        if isinstance(event, rfxtrxmod.ConnectionLost):
+            _LOGGER.warning("Connection was lost, triggering reload")
+            hass.async_create_task(
+                hass.config_entries.async_reload(entry.entry_id),
```

Comment:
We have `hass.config_entries.async_schedule_reload` now.

https://github.com/home-assistant/core/blob/9e2db708b9015758e174bff3a07e8e7bfb85faf0/homeassistant/config_entries.py#L1576